Automattic/wp-calypso
# composite-checkout: Refactor ManagedContactDetails

#### Changes proposed in this Pull Request

* Refactors the `ManagedContactDetails` type and adds ccTLD data. Pulled out of #40482 for independent review.

#### Testing instructions

```
npm run test-client client/my-sites/checkout/composite-checkout/wpcom/test/wpcom-store-state.ts
```

This should have no observable effect on checkout, but it is a fairly significant refactoring of the state data type, so it will be important to make sure nothing is broken.
